src/test/java/core/basesyntax/MainTest.java

```diff
@@ -0,0 +1,366 @@
+    private static OperationStrategy operationStrategy;
+    private static DataProcessor dataProcessor;
+    private static AmountHandler additionHandler;
+    private static AmountHandler subtractionHandler;
+    private static FileReader fileReaderImpl;
+    private static ReportWriter reportWriter;
+    private static RowParser rowParser;
+    private static DataValidator dataValidator;
+    private static FruitDao fruitDao;
+
+    @Before
+    public void setUp() {
+        additionHandler = new AdditionHandler();
+        subtractionHandler = new SubtractionHandler();
+        fileReaderImpl = new FileReaderImpl();
+        reportWriter = new ReportWriterImpl();
+        rowParser = new RowParserImpl();
+        dataValidator = new DataValidatorImpl();
+        fruitDao = new FruitDaoImpl();
+        fruitRecords = new ArrayList<>();
+        fruitRecords.add(new FruitRecord(20, BALANCE, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(100, BALANCE, new Fruit("apple")));
+        fruitRecords.add(new FruitRecord(100, SUPPLY, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(13, PURCHASE, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(10, RETURN, new Fruit("apple")));
+        fruitRecords.add(new FruitRecord(20, PURCHASE, new Fruit("apple")));
+        fruitRecords.add(new FruitRecord(5, PURCHASE, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(50, SUPPLY, new Fruit("banana")));
+        reportRecords = new ArrayList<>();
+        reportRecords.add("banana,152");
+        reportRecords.add("apple,90");
+        strategies = new HashMap<>();
+        strategies.put(BALANCE, additionHandler);
+        strategies.put(SUPPLY, additionHandler);
+        strategies.put(RETURN, additionHandler);
+        strategies.put(PURCHASE, subtractionHandler);
+        operationStrategy = new OperationStrategyImpl(strategies);
+        dataProcessor = new DataProcessorImpl();
+        fileData = new ArrayList<>();
+        fileData.add("b,banana,20");
+        fileData.add("b,apple,100");
+        fileData.add("s,banana,100");
+        fileData.add("p,banana,13");
+        fileData.add("r,apple,10");
+        fileData.add("p,apple,20");
+        fileData.add("p,banana,5");
+        fileData.add("s,banana,50");
+    }
+
+    @Test
+    public void getCorrectAmountHandler_OK() {
+        AmountHandler expectedBalance = strategies.get(BALANCE);
+        AmountHandler expectedSupply = strategies.get(SUPPLY);
+        AmountHandler expectedReturn = strategies.get(RETURN);
+        AmountHandler expectedPurchase = strategies.get(PURCHASE);
+        AmountHandler actualBalance = operationStrategy.get(BALANCE);
+        AmountHandler actualSupply = operationStrategy.get(SUPPLY);
+        AmountHandler actualReturn = operationStrategy.get(RETURN);
+        AmountHandler actualPurchase = operationStrategy.get(PURCHASE);
+        Assert.assertEquals("Test failed! For operation type "
```

Comment:
you can test balance, purchase, return, supply, in separate methods for better visibility when smth goes wrong